frontend/src/metabase/query_builder/components/DatasetEditor/utils.ts

```diff
@@ -0,0 +1,12 @@
+import { getIconForField } from "metabase/lib/schema_metadata";
+
+type Column = {
+  semantic_type?: string;
+};
+
+export function getColumnTypeIcon(column?: Column): string {
```

Comment:
Can you move this to a more general utils file and rename the function to `getSemanticTypeIcon` or similar? I actually need this function for something I'm working on 😛

Reply:
also, feel free to merge, and I can take care of this in my PR

Reply:
@daltojohnso I'd be happy to, but I should warn you this PR targets a feature branch and there is a bigger PR (#19724) to be merged there before this code will land `master`. If you need it sooner, go for it and I'll refactor the editor code to use your utility

Reply:
I'm digging more into this and wondering if instead of using `getIconForField` we use the data declared in `frontend/src/metabase/lib/core.js`. I imagine we'd want to add more icons for specific semantic types over time, so perhaps we can add an `icon` property to items in the `field_semantic_types` collection and then rely on that directly with a function like `getSemanticTypeIcon`